pkg/core/user/context.go

```diff
@@ -0,0 +1,16 @@
+package user
+
+import "context"
+
+type userCtx struct{}
+
+func UserCtx(ctx context.Context, user User) context.Context {
```

Comment:
Package is already named `user` I don't think we have to start functions with `User` prefix